<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>(EdgeR) WT vs sxy, Time = 0</title>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>

<!-- MathJax scripts -->
<script type="text/javascript" src="https://c328740.ssl.cf1.rackcdn.com/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>


<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}

pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}
</style>



</head>

<body>
<h1>(EdgeR) WT vs sxy, Time = 0</h1>

<h2>Preprocessing Code</h2>

<pre><code class="r"># Main Pipeline Using EdgeR (WT vs sxy), at time = 0.  by Nathaniel (April
# 2014)

library(limma)
library(edgeR)

# Reset Workspace
rm(list = ls())

# Loads RNA-Seq Raw Fragment Count
raw.reads &lt;- read.table(&quot;Data/fragment_counts.txt&quot;, header = TRUE, sep = &quot;\t&quot;)

# Loads Design Matrix for limma
frag.matrix &lt;- read.table(&quot;Data/FragmentMatrix.txt&quot;, header = TRUE, sep = &quot;\t&quot;)

# Loads Truncated GeneList
gene.list &lt;- read.table(&quot;Data/GeneID.txt&quot;, header = TRUE)

# Loads crp-S Target List
crp.list &lt;- read.table(&quot;Data/crp.txt&quot;, header = TRUE)

# Associating GeneID as KEY identifier for all datasets
rownames(gene.list) &lt;- gene.list$ID
rownames(raw.reads) &lt;- gene.list$ID

# Creating Data Subset (with Restructuring) Wild Type and Sxy (K and S);
# Time = 0
small.meta &lt;- subset(frag.matrix, frag.matrix$Type %in% c(&quot;K&quot;, &quot;S&quot;) &amp; frag.matrix$Timepoint %in% 
    &quot;0&quot;)
small.meta$Type &lt;- factor(small.meta$Type)
small.raw &lt;- raw.reads[, colnames(raw.reads) %in% small.meta$Sample]

# Unloading Unneeded Files
rm(raw.reads)
rm(frag.matrix)

# Running Dataset through EdgeR pipeline 1. Creating DGEList from Trimmed
# Data
dge.list &lt;- DGEList(counts = small.raw, group = small.meta$Type)

# 2. Creating Design Matrix
sm.design &lt;- model.matrix(~Type, small.meta)

# 3. Continuing EdgeR Pipeline
dge.common.list &lt;- estimateGLMCommonDisp(dge.list, sm.design)
dge.trend.list &lt;- estimateGLMTrendedDisp(dge.common.list, sm.design)
dge.tag.list &lt;- estimateGLMTagwiseDisp(dge.trend.list, sm.design)

# 4. Fitting and Throwing Out Hits **Current Null Hypothesis: TypeS (i.e. WT
# against sxy at time = 0)
dge.fit &lt;- glmFit(dge.tag.list, sm.design)
dge.lrt &lt;- glmLRT(dge.fit, coef = &quot;TypeS&quot;)
dge.top &lt;- topTags(dge.lrt, n = Inf)
dge.cpm &lt;- cpm(dge.tag.list)

final.top.table &lt;- dge.top$table[dge.top$table$FDR &lt; 0.05, ]
final.top.cpm &lt;- dge.cpm[rownames(final.top.table), ]
final.top.genes &lt;- gene.list[gene.list$ID %in% rownames(final.top.table), ]

# Creating crp-S Subset List
crps.list &lt;- subset(crp.list, prom %in% c(&quot;crps&quot;, &quot;sxy&quot;))
rownames(crps.list) &lt;- NULL

# Creating crp-N Subset List
crpn.list &lt;- subset(crp.list, prom %in% &quot;crpn&quot;)
rownames(crpn.list) &lt;- NULL
</code></pre>

<h2>Analysis Results</h2>

<h3>Number of differentially expressed and up/down-regulated genes</h3>

<pre><code class="r"># Number of differentially expressed genes:
nrow(final.top.genes)
</code></pre>

<pre><code>## [1] 53
</code></pre>

<pre><code class="r">
# Number of down-regulated genes:
dge.down &lt;- final.top.table[final.top.table$logFC &lt; 0, ]
nrow(dge.down)
</code></pre>

<pre><code>## [1] 49
</code></pre>

<pre><code class="r">
# Number of up-regulated genes:
dge.up &lt;- final.top.table[final.top.table$logFC &gt; 0, ]
nrow(dge.up)
</code></pre>

<pre><code>## [1] 4
</code></pre>

<h3>crp-S Genes: Number of differentially, non-differentially and up/down-regulated genes</h3>

<pre><code class="r"># crp-S genes: Number of differentially expressed genes:
crps.diff &lt;- final.top.table[rownames(final.top.table) %in% crps.list$orf, ]
nrow(crps.diff)
</code></pre>

<pre><code>## [1] 1
</code></pre>

<pre><code class="r">
# Number of non-differentially expressed genes:
tmp.crps.non &lt;- crps.list[-which(crps.list$orf %in% rownames(crps.diff)), ]
crps.non &lt;- dge.top$table[as.character(tmp.crps.non$orf), ]
rm(tmp.crps.non)
nrow(crps.non)
</code></pre>

<pre><code>## [1] 26
</code></pre>

<pre><code class="r">
# Number of down-regulated genes:
crps.down &lt;- crps.diff[crps.diff$logFC &lt; 0, ]
nrow(crps.down)
</code></pre>

<pre><code>## [1] 1
</code></pre>

<pre><code class="r">
# Number of up-regulated genes:
crps.up &lt;- crps.diff[crps.diff$logFC &gt; 0, ]
nrow(crps.up)
</code></pre>

<pre><code>## [1] 0
</code></pre>

<h3>crp-N Genes: Number of differentially, non-differentially and up/down-regulated genes</h3>

<pre><code class="r"># crp-N genes: Number of differentially expressed genes:
crpn.diff &lt;- final.top.table[rownames(final.top.table) %in% crpn.list$orf, ]
nrow(crpn.diff)
</code></pre>

<pre><code>## [1] 14
</code></pre>

<pre><code class="r">
# Number of non-differentially expressed genes:
tmp.crpn.non &lt;- crpn.list[-which(crpn.list$of %in% rownames(crpn.diff)), ]
crpn.non &lt;- dge.top$table[as.character(tmp.crpn.non$orf), ]
rm(tmp.crpn.non)
nrow(crpn.non)
</code></pre>

<pre><code>## [1] 0
</code></pre>

<pre><code class="r">
# Number of down-regulated genes:
crpn.down &lt;- crpn.diff[crpn.diff$logFC &lt; 0, ]
nrow(crpn.down)
</code></pre>

<pre><code>## [1] 14
</code></pre>

<pre><code class="r">
# Number of up-regulated genes:
crpn.up &lt;- crpn.diff[crpn.diff$logFC &gt; 0, ]
nrow(crpn.up)
</code></pre>

<pre><code>## [1] 0
</code></pre>

<h3>All differentially expressed genes</h3>

<pre><code class="r"># Displaying all differentially expressed genes
show(final.top.genes)
</code></pre>

<pre><code>##                  ID
## HI_0062     HI_0062
## HI_0126     HI_0126
## HI_0129     HI_0129
## HI_0131     HI_0131
## HI_0134     HI_0134
## HI_0146     HI_0146
## HI_0147     HI_0147
## HI_0148     HI_0148
## HI_0206     HI_0206
## HI_0262     HI_0262
## HI_0263     HI_0263
## HI_0264     HI_0264
## HI_0287     HI_0287
## HI_0288     HI_0288
## HI_0289     HI_0289
## HI_0501     HI_0501
## HI_0520     HI_0520
## HI_0521     HI_0521
## HI_0534     HI_0534
## HI_0601     HI_0601
## HI_0620     HI_0620
## HI_0684     HI_0684
## HI_0712     HI_0712
## HI_0745     HI_0745
## HI_0809     HI_0809
## HI_0822     HI_0822
## HI_0823     HI_0823
## HI_0824     HI_0824
## HI_1010     HI_1010
## HI_1011     HI_1011
## HI_1012     HI_1012
## HI_1013     HI_1013
## HI_1014     HI_1014
## HI_1015     HI_1015
## HI_1016     HI_1016
## HI_1020     HI_1020
## HI_1080     HI_1080
## HI_1126.1 HI_1126.1
## HI_1127     HI_1127
## HI_1173     HI_1173
## HI_1218     HI_1218
## HI_1245     HI_1245
## HI_1356     HI_1356
## HI_1357     HI_1357
## HI_1398     HI_1398
## HI_1427     HI_1427
## HI_1434.1 HI_1434.1
## HI_1505     HI_1505
## HI_1506     HI_1506
## HI_1518     HI_1518
## HI_1586     HI_1586
## HI_1659     HI_1659
## HI_1712     HI_1712
##                                                                      Names
## HI_0062                                     dnaK suppressor protein (dksA)
## HI_0126               ferric ABC transporter%2C ATP-binding protein (afuC)
## HI_0129                                     predicted coding region HI0129
## HI_0131                                                       afuA protein
## HI_0134                                     predicted coding region HI0134
## HI_0146                                     conserved hypothetical protein
## HI_0147                       conserved hypothetical transmembrane protein
## HI_0148                                     conserved hypothetical protein
## HI_0206                                        5&#39;-nucleotidase%2C putative
## HI_0262                               heme-hemopexin utilization protein C
## HI_0263                        heme-hemopexin utilization protein B (hxuB)
## HI_0264                        heme-hemopexin utilization protein A (hxuA)
## HI_0287                        tryptophan-specific transport protein (mtr)
## HI_0288                                          L-serine deaminase (sdaA)
## HI_0289                                          serine transporter (sdaC)
## HI_0501                      high affinity ribose transport protein (rbsD)
## HI_0520                                     conserved hypothetical protein
## HI_0521                                     conserved hypothetical protein
## HI_0534                                     aspartate ammonia-lyase (aspA)
## HI_0601                                  DNA transformation protein (tfoX)
## HI_0620                               28 kDa outer membrane protein (hlpA)
## HI_0684   anaerobic glycerol-3-phosphate dehydrogenase%2C subunit B (glpB)
## HI_0712                                         hemoglobin-binding protein
## HI_0745                                           L-asparaginase II (ansB)
## HI_0809                           phosphoenolpyruvate carboxykinase (pckA)
## HI_0822    galactose ABC transporter%2C periplasmic-binding protein (mglB)
## HI_0823          galactoside ABC transporter%2C ATP-binding protein (mglA)
## HI_0824             galactoside ABC transporter%2C permease protein (mglC)
## HI_1010                     3-hydroxyisobutyrate dehydrogenase%2C putative
## HI_1011                                     conserved hypothetical protein
## HI_1012                                        sugar isomerase%2C putative
## HI_1013                                     conserved hypothetical protein
## HI_1014                                     conserved hypothetical protein
## HI_1015                                          gluconate permease (gntP)
## HI_1016                                     predicted coding region HI1016
## HI_1020            thiamin ABC transporter%2C permease protein%2C putative
## HI_1080          amino acid ABC transporter%2C periplasmic-binding protein
## HI_1126.1                                 predicted coding region HI1126.1
## HI_1127                                     predicted coding region HI1127
## HI_1173                                                sprt protein (sprT)
## HI_1218                                          L-lactate permease (lctP)
## HI_1245                                  malate oxidoreductase%2C putative
## HI_1356                                  4-alpha-glucanotransferase (malQ)
## HI_1357                         1%2C4-alpha-glucan branching enzyme (glgB)
## HI_1398                              fumarate hydratase%2C class II (fumC)
## HI_1427                                     predicted coding region HI1427
## HI_1434.1                                   cold shock-like protein (cspD)
## HI_1505                                     predicted coding region HI1505
## HI_1506                                     predicted coding region HI1506
## HI_1518                                     predicted coding region HI1518
## HI_1586                   conserved hypothetical integral membrane protein
## HI_1659         ribonucleoside-diphosphate reductase%2C alpha chain (nrdA)
## HI_1712              phosphoenolpyruvate-protein phosphotransferase (ptsI)
</code></pre>

<h3>All down-regulated genes</h3>

<pre><code class="r"># Displaying all down-regulated genes
dge.down.genes &lt;- gene.list[gene.list$ID %in% rownames(dge.down), ]
show(dge.down.genes)
</code></pre>

<pre><code>##                  ID
## HI_0126     HI_0126
## HI_0129     HI_0129
## HI_0131     HI_0131
## HI_0134     HI_0134
## HI_0146     HI_0146
## HI_0147     HI_0147
## HI_0148     HI_0148
## HI_0206     HI_0206
## HI_0262     HI_0262
## HI_0263     HI_0263
## HI_0264     HI_0264
## HI_0287     HI_0287
## HI_0288     HI_0288
## HI_0289     HI_0289
## HI_0501     HI_0501
## HI_0520     HI_0520
## HI_0521     HI_0521
## HI_0534     HI_0534
## HI_0601     HI_0601
## HI_0620     HI_0620
## HI_0684     HI_0684
## HI_0712     HI_0712
## HI_0745     HI_0745
## HI_0809     HI_0809
## HI_0822     HI_0822
## HI_0823     HI_0823
## HI_0824     HI_0824
## HI_1010     HI_1010
## HI_1011     HI_1011
## HI_1012     HI_1012
## HI_1013     HI_1013
## HI_1014     HI_1014
## HI_1015     HI_1015
## HI_1016     HI_1016
## HI_1020     HI_1020
## HI_1080     HI_1080
## HI_1126.1 HI_1126.1
## HI_1127     HI_1127
## HI_1218     HI_1218
## HI_1245     HI_1245
## HI_1356     HI_1356
## HI_1357     HI_1357
## HI_1398     HI_1398
## HI_1427     HI_1427
## HI_1434.1 HI_1434.1
## HI_1505     HI_1505
## HI_1506     HI_1506
## HI_1659     HI_1659
## HI_1712     HI_1712
##                                                                      Names
## HI_0126               ferric ABC transporter%2C ATP-binding protein (afuC)
## HI_0129                                     predicted coding region HI0129
## HI_0131                                                       afuA protein
## HI_0134                                     predicted coding region HI0134
## HI_0146                                     conserved hypothetical protein
## HI_0147                       conserved hypothetical transmembrane protein
## HI_0148                                     conserved hypothetical protein
## HI_0206                                        5&#39;-nucleotidase%2C putative
## HI_0262                               heme-hemopexin utilization protein C
## HI_0263                        heme-hemopexin utilization protein B (hxuB)
## HI_0264                        heme-hemopexin utilization protein A (hxuA)
## HI_0287                        tryptophan-specific transport protein (mtr)
## HI_0288                                          L-serine deaminase (sdaA)
## HI_0289                                          serine transporter (sdaC)
## HI_0501                      high affinity ribose transport protein (rbsD)
## HI_0520                                     conserved hypothetical protein
## HI_0521                                     conserved hypothetical protein
## HI_0534                                     aspartate ammonia-lyase (aspA)
## HI_0601                                  DNA transformation protein (tfoX)
## HI_0620                               28 kDa outer membrane protein (hlpA)
## HI_0684   anaerobic glycerol-3-phosphate dehydrogenase%2C subunit B (glpB)
## HI_0712                                         hemoglobin-binding protein
## HI_0745                                           L-asparaginase II (ansB)
## HI_0809                           phosphoenolpyruvate carboxykinase (pckA)
## HI_0822    galactose ABC transporter%2C periplasmic-binding protein (mglB)
## HI_0823          galactoside ABC transporter%2C ATP-binding protein (mglA)
## HI_0824             galactoside ABC transporter%2C permease protein (mglC)
## HI_1010                     3-hydroxyisobutyrate dehydrogenase%2C putative
## HI_1011                                     conserved hypothetical protein
## HI_1012                                        sugar isomerase%2C putative
## HI_1013                                     conserved hypothetical protein
## HI_1014                                     conserved hypothetical protein
## HI_1015                                          gluconate permease (gntP)
## HI_1016                                     predicted coding region HI1016
## HI_1020            thiamin ABC transporter%2C permease protein%2C putative
## HI_1080          amino acid ABC transporter%2C periplasmic-binding protein
## HI_1126.1                                 predicted coding region HI1126.1
## HI_1127                                     predicted coding region HI1127
## HI_1218                                          L-lactate permease (lctP)
## HI_1245                                  malate oxidoreductase%2C putative
## HI_1356                                  4-alpha-glucanotransferase (malQ)
## HI_1357                         1%2C4-alpha-glucan branching enzyme (glgB)
## HI_1398                              fumarate hydratase%2C class II (fumC)
## HI_1427                                     predicted coding region HI1427
## HI_1434.1                                   cold shock-like protein (cspD)
## HI_1505                                     predicted coding region HI1505
## HI_1506                                     predicted coding region HI1506
## HI_1659         ribonucleoside-diphosphate reductase%2C alpha chain (nrdA)
## HI_1712              phosphoenolpyruvate-protein phosphotransferase (ptsI)
</code></pre>

<h3>All up-regulated genes</h3>

<pre><code class="r"># Displaying all up-regulated genes
dge.up.genes &lt;- gene.list[gene.list$ID %in% rownames(dge.up), ]
show(dge.up.genes)
</code></pre>

<pre><code>##              ID                                            Names
## HI_0062 HI_0062                   dnaK suppressor protein (dksA)
## HI_1173 HI_1173                              sprt protein (sprT)
## HI_1518 HI_1518                   predicted coding region HI1518
## HI_1586 HI_1586 conserved hypothetical integral membrane protein
</code></pre>

</body>

</html>
